### PR TITLE
trivial: Add daemon configuration key 'OnlyTrusted' to dbus Properties

### PR DESCRIPTION
**Pull request by CragW**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupd/fwupd-client.c [hunk 1]
```diff
@@ -3280,6 +3294,24 @@ fwupd_client_get_tainted(FwupdClient *self)
 	return priv->tainted;
 }
 
+/**
+ * fwupd_client_get_only_trusted:
+ * @self: a #FwupdClient
+ *
+ * Gets if the daemon is verifying lvfs sign for firmware files.
```
Comment (superm1) [inline libfwupd/fwupd-client.c:3301]: Verifying signatures from a trusted authority. LVFS isn't the only authority

File: libfwupd/fwupd-client.c [hunk 2]
```diff
@@ -3280,6 +3294,24 @@ fwupd_client_get_tainted(FwupdClient *self)
...
+ *
+ * Returns: %TRUE if the daemon is checking lvfs sign
```
Comment (superm1) [inline libfwupd/fwupd-client.c:3303]: Checking signatures.

File: src/org.freedesktop.fwupd.xml [hunk 3]
```diff
@@ -111,6 +111,17 @@
       </doc:doc>
     </property>
 
+    <!--***********************************************************-->
+    <property name='OnlyTrusted' type='b' access='read'>
+      <doc:doc>
+        <doc:description>
+          <doc:para>
+            If the daemon is verifying lvfs signed firmware.
```
Comment (superm1) [inline src/org.freedesktop.fwupd.xml:119]: Daemon requires trusted payloads

**Review by superm1 — COMMENTED** [review]
Overall looks good, just some nits.

**Review by superm1 — APPROVED** [review]
This lgtm. I think a logical use of this would be to show a warning on the client that the Daemon is requiring verifying signatures at the time too. @hughsie thoughts on that?

File: libfwupd/fwupd-client.c [hunk 4]
```diff
@@ -421,6 +423,14 @@ fwupd_client_properties_changed_cb(GDBusProxy *proxy,
 		if (val != NULL)
 			fwupd_client_set_host_security_id(self, g_variant_get_string(val, NULL));
 	}
+	if (g_variant_dict_contains(dict, "OnlyTrusted")) {
+		g_autoptr(GVariant) val = NULL;
+		val = g_dbus_proxy_get_cached_property(proxy, "OnlyTrusted");
+		if (val != NULL) {
+			priv->only_trusted = g_variant_get_boolean(val);
+			fwupd_client_object_notify(self, "onlytrusted");
```
Comment (hughsie) [inline libfwupd/fwupd-client.c:431]: this needs to be `only-trusted` -- I've fixed it up for you.

File: libfwupd/fwupd-client.c [hunk 5]
```diff
@@ -5414,6 +5449,20 @@ fwupd_client_class_init(FwupdClientClass *klass)
...
 	g_object_class_install_property(object_class, PROP_HOST_SECURITY_ID, pspec);
+
+	/**
+	 * FwupdClient:only_trusted:
+	 *
+	 * If the daemon is verifying lvfs signed firmware.
+	 *
+	 * Since: 1.8.0
+	 */
+	pspec = g_param_spec_boolean("onlytrusted",
```
Comment (hughsie) [inline libfwupd/fwupd-client.c:5460]: same, with a `-` -- fixed.

File: src/fu-engine.c [hunk 6]
```diff
@@ -6274,6 +6275,13 @@ fu_engine_get_tainted(FuEngine *self)
 	return self->tainted;
 }
 
+gboolean
+fu_engine_get_only_trusted(FuEngine *self)
+{
+	g_return_val_if_fail(FU_IS_ENGINE(self), TRUE);
```
Comment (hughsie) [inline src/fu-engine.c:6281]: we always return FALSE or NULL from these, fixed for you.

**Comment by hughsie** [comment]
> show a warning on the client that the daemon is not requiring verifying signatures

I think that's a really good idea, but we can follow this idea up in a new PR.

**Review by hughsie — APPROVED** [review]

**Pull request merged**